src/core/shared_memory_manager.cc

```diff
@@ -205,6 +205,20 @@ SharedMemoryManager::UnregisterAllSharedMemory()
   return Status::Success;
 }
 
+Status
+SharedMemoryManager::GetSharedMemoryStatus(SharedMemoryStatus* shm_status)
+{
+  for (const auto& shm_info : shared_memory_map_) {
```

Comment:
Need lock to protect shared_memory_map_.  Check all the code to make sure that datastructures are protected appropriately.

Reply:
Done